frontend/test/pytest/test_jax_dynamic_api.py

```diff
@@ -936,6 +1013,31 @@ def case():
     assert_array_and_dtype_equal(f(False, a, b), jnp.zeros(3))
 
 
+def test_qnode_cond_capture():
+    """Test that catalyst tensor primitive is compatible with quantum conditional"""
+
+    @qjit
+    @qml.qnode(qml.device("lightning.qubit", wires=4))
+    def f(flag, sz):
+        a = jnp.ones([sz, 3], dtype=float)
+
+        @cond(flag)
+        def case():
+            b = jnp.ones([sz, 3], dtype=float)
+            return a + b
+
+        @case.otherwise
+        def case():
+            b = jnp.ones([sz, 3], dtype=float)
```

Comment:
Change to zeros just to have differences in the paths.

Reply:
Done.